tools/NewModuleScript/MainScreenDialogTemplate.kt

```diff
@@ -0,0 +1,129 @@
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.lifecycle.viewmodel.compose.viewModel
+import com.arcgismaps.toolkit.geoviewcompose.MapView
+import com.esri.arcgismaps.sample.displaycomposablemapview.components.MapViewModel
+import com.esri.arcgismaps.sample.sampleslib.components.DropDownMenuBox
+import com.esri.arcgismaps.sample.sampleslib.components.MessageDialog
+import com.esri.arcgismaps.sample.sampleslib.components.SampleDialog
+import com.esri.arcgismaps.sample.sampleslib.components.SamplePreviewSurface
+import com.esri.arcgismaps.sample.sampleslib.components.SampleTopAppBar
+
+/**
+ * Main screen layout for the sample app
+ */
+@Composable
+fun MainScreen(sampleName: String) {
+    val mapViewModel: MapViewModel = viewModel()
+    var isDialogOptionsVisible by remember { mutableStateOf(false) }
+
+    Scaffold(
+        topBar = { SampleTopAppBar(title = sampleName) },
+        floatingActionButton = {
+            if (!isDialogOptionsVisible) {
+                FloatingActionButton(
+                    modifier = Modifier.padding(bottom = 36.dp, end = 12.dp),
+                    onClick = { isDialogOptionsVisible = true }
+                ) { Icon(Icons.Filled.Settings, contentDescription = "Show options") }
+            }
+        },
+        content = {
+            Column(
+                modifier = Modifier
+                    .fillMaxSize()
+                    .padding(it),
+            ) {
+                MapView(
+                    modifier = Modifier
+                        .fillMaxSize()
+                        .weight(1f),
+                    arcGISMap = mapViewModel.arcGISMap
+                )
+            }
+
+            if (isDialogOptionsVisible) {
+                DialogOptions(
+                    // isCurrentOptionEnabled = ...,
+                    // onOptionToggled = { ... }
+                    onDismissRequest = { isDialogOptionsVisible = false }
+                )
+            }
+
+            mapViewModel.messageDialogVM.apply {
+                if (dialogStatus) {
+                    MessageDialog(
+                        title = messageTitle,
+                        description = messageDescription,
+                        onDismissRequest = ::dismissDialog
+                    )
+                }
+            }
+        }
+    )
+}
+
+@Composable
+fun DialogOptions(
+    onDismissRequest: () -> Unit
+) {
+    SampleDialog(onDismissRequest = onDismissRequest) {
+        Text("Sample options: ", style = MaterialTheme.typography.titleMedium)
+        DropDownMenuBox(
+            textFieldValue = "<selected-option>",
+            textFieldLabel = "Select an option",
+            dropDownItemList = emptyList(),
```

Comment:
I think it would be good to have a couple of dummy options here.

Reply:
I am a little hesitant in using _dummy_ options in a `list` as a sample COULD use it in this manner but ideally would map a list of items or enums provided from a `viewmodel`. Visit [sample usages](https://github.com/Esri/arcgis-maps-sdk-kotlin-samples/blob/main/samples/manage-features/src/main/java/com/esri/arcgismaps/sample/managefeatures/screens/ManageFeaturesScreen.kt#L127) of these components to find helpful examples.